tests/scripts/check_names.py

```diff
@@ -0,0 +1,830 @@
+        typo_exclusion = re.compile(r"XXX|__|_$|^MBEDTLS_.*CONFIG_FILE$")
+
+        for name_match in self.parse_result["mbed_words"]:
+            found = name_match.name in all_caps_names
+
+            # Since MBEDTLS_PSA_ACCEL_XXX defines are defined by the
+            # PSA driver, they will not exist as macros. However, they
+            # should still be checked for typos using the equivalent
+            # BUILTINs that exist.
+            if "MBEDTLS_PSA_ACCEL_" in name_match.name:
+                found = name_match.name.replace(
+                    "MBEDTLS_PSA_ACCEL_",
+                    "MBEDTLS_PSA_BUILTIN_") in all_caps_names
+
+            if not found and not typo_exclusion.search(name_match.name):
+                problems.append(Typo(name_match))
+
+        self.output_check_result(quiet, "Likely typos", problems)
+        return len(problems)
+
+    def output_check_result(self, quiet, name, problems):
+        """
+        Write out the PASS/FAIL status of a performed check depending on whether
+        there were problems.
+
+        Args:
+        * quiet: whether to hide detailed problem explanation.
+        * name: the name of the test
+        * problems: a List of encountered Problems
+        """
+        if problems:
+            self.log.info("{}: FAIL\n".format(name))
+            for problem in problems:
+                problem.quiet = quiet
+                self.log.warning(str(problem))
+        else:
+            self.log.info("{}: PASS".format(name))
+
+def main():
+    """
+    Perform argument parsing, and create an instance of CodeParser and
+    NameChecker to begin the core operation.
+    """
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        description=(
+            "This script confirms that the naming of all symbols and identifiers "
+            "in Mbed TLS are consistent with the house style and are also "
+            "self-consistent.\n\n"
+            "Expected to be run from the MbedTLS root directory.")
+    )
+    parser.add_argument(
+        "-v", "--verbose",
+        action="store_true",
+        help="show parse results"
+    )
+    parser.add_argument(
+        "-q", "--quiet",
+        action="store_true",
+        help="hide unnecessary text, explanations, and highlighs"
+    )
+
+    args = parser.parse_args()
+
+    # Configure the global logger, which is then passed to the classes below
+    log = logging.getLogger()
+    log.setLevel(logging.DEBUG if args.verbose else logging.INFO)
+    log.addHandler(logging.StreamHandler())
+
+    try:
+        code_parser = CodeParser(log)
+        parse_result = code_parser.comprehensive_parse()
+    except Exception: # pylint: disable=broad-except
+        traceback.print_exc()
+        sys.exit(2)
+
+    name_checker = NameChecker(parse_result, log)
+    return_code = name_checker.perform_checks(quiet=args.quiet)
```

Comment:
the `quiet` argument is passed around, until it reaches the `Problem`s child instance.
This argument will never change it's value once set, there is no point in passing it around.
The `Problem` class instances should be already created with the correctly set `quiet` attribute. You could implement a factory pattern for that but it would require a bit more effort.

Alternatively, you could utilize the __class__ mechanism, which "sort of" works like a factory:
```
from abc import ABC, abstractmethod

class Problem(ABC):
    def __str__(self):
        if self.__class__.quiet:  # this is a bit hacky, but requires least effort for the current state of the code
           return self.quiet_output()
        return self.verbal_output()

    @abstractmethod
    def quiet_output(self):
        pass

    @abstractmethod
    def verbal_output(self):
        pass

    @classmethod
    def set_quiet(cls, quiet): # this is a bit hacky, but requires least effort for the current state of the code
        cls.quiet = quiet

class Child(Problem):
    def quiet_output(self):
        return "child quiet output"

    def verbal_output(self):
        return "child verbal output"

Problem.set_quiet(True) #modifies parent class
c_quiet = Child() 
print(c_quiet)

Problem.set_quiet(False) #modifies parent class
c_verbal = Child()
print(c_verbal)

parent = Problem() # raises error, can't implement abstract class (assuming that this is what you want)
```
This should allow you to create and use the objects deriving from `Problem` without needing to pass the `quiet` option around the functions.

Reply:
Sounds great. Thanks, didn't know Python had proper support for abstract classes. I'll do this.

Reply:
I realised I could achieve the same thing by using class variables (modifying the parent class definition), which has the same behaviour without the boilerplate of ABCs, so I've gone with that approach. Please take a look and confirm I haven't made a mistake in my understanding!

Reply:
The proposed piece of code was ment to address both problems I mentioned:
- passing around the `quiet` variable
- copying the `if quiet: print("something") ... else: print("something else")` logic to overriden `__str__` methods.

You have fixed the first issue. The second still remains.